tsk/fs/ntfs.c

```diff
@@ -2600,11 +2608,31 @@ ntfs_proc_attrlist(NTFS_INFO * ntfs,
         /* Process the attribute seq for this MFT entry and add them
          * to the TSK_FS_META structure
          */
+        if (processed_inum_list != NULL && tsk_stack_find(processed_inum_list, mftToDo[a])) {
+            tsk_error_reset();
+            tsk_error_set_errno(TSK_ERR_FS_CORRUPT);
+            tsk_error_set_errstr("ntfs_proc_attrlist: MFT %" PRIuINUM
+                " seen in more than one attribute list for %"
+                PRIuINUM
+                " (base file ref = %" PRIuINUM ")",
+                mftToDo[a],
+                fs_file->meta->addr,
+                tsk_getu48(fs->endian, mft->base_ref));
+            free(mft);
+            free(map);
+            free(buf);
+            return TSK_COR;
+        }
+
+        if (processed_inum_list == NULL) {
+            processed_inum_list = mftSeenList = tsk_stack_create();
```

Comment:
This could use a comment. I believe mftSeenList is being used to check if the stack was created in this method so we know whether to free it?

Reply:
Yes, that's right. I've added a comment to the code.